### PR TITLE
Generate package within 63 character limit when creating function

### PR DESCRIPTION
**Pull request by shubham-bansal96**
## Description
While creating a package if package name is more than 63 characters, then package get failed.
This fix will allow to create a package even if package name is more than 63 characters by taking first 26 characters as package name and appending unique id to it, and will trim rest of the characters of package name.

## Which issue(s) this PR fixes:

Fixes # https://fissionio.slack.com/archives/C3LUX6BBP/p1657148131481119

## Testing
Created a simple function with function name less than 63 characters.
Created a function with function name more than 63 characters.
Created a function using spec with function name less than 63 characters.
Created a function using spec with function name more than 63 characters.
Created a package manually with package name less than 63 characters.
Created a package manually with package name more than 63 characters.

## Checklist:

- [x] I ran tests as well as code linting locally to verify my changes. 
- [x] I have done manual verification of my changes, changes working as expected.
- [x] I have added new tests to cover my changes.
- [x] My changes follow contributing guidelines of Fission.
- [ ] I have signed all of my commits.

### CODE REVIEW

**Review by sanketsudake — CHANGES_REQUESTED** [review]

File: pkg/fission-cli/cmd/function/create_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	fnName string
+}
+
+func TestGeneratePackageName(t *testing.T) {
+	for _, test := range []struct {
+		name     string
+		request  *pkgRequest
+		expected int
+	}{
+		{
```
Comment (sanketsudake) [inline pkg/fission-cli/cmd/function/create_test.go:16]: Also add tests cases for normal names and smaller length function names.

File: pkg/fission-cli/cmd/function/create_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+			name: "name of package should be less than or equal to 63 characters, if function name is less than or equal to 26 characters",
+			request: &pkgRequest{
+				uid:    "35764439-0676-48bf-bb61-c3cdcc82b801",
```
Comment (sanketsudake) [inline pkg/fission-cli/cmd/function/create_test.go:19]: Instead of hardcoding we can generate UUID compatible to one we are getting in generate function.

**Review by sanketsudake — CHANGES_REQUESTED** [review]

File: pkg/fission-cli/cmd/function/create_test.go (hunk 2 above)
Comment (shubham-bansal96) [reply pkg/fission-cli/cmd/function/create_test.go]: done

File: pkg/fission-cli/cmd/function/create.go [hunk 3]
```diff
@@ -324,6 +324,15 @@ func (opts *CreateSubCommand) complete(input cli.Input) error {
 	return nil
 }
 
+// generatePackgeName => will return package name by appending id in function name and will make sure that package name will never be more than length of 63 characters.
+func generatePackageName(fnName string, id string) string {
+	if len(fnName) <= 26 {
```
[the remark this answers is not in the document]
Comment (shubham-bansal96) [reply pkg/fission-cli/cmd/function/create.go]: done

File: pkg/fission-cli/cmd/package/create.go [hunk 4]
```diff
@@ -62,6 +62,12 @@ func (opts *CreateSubCommand) run(input cli.Input) error {
 			console.Warn(fmt.Sprintf("--%v will be soon marked as required flag, see 'help' for details", flagkey.HtName))
 		}
 	}
+
+	if len(pkgName) > 63 {
```
[the remark this answers is not in the document]
Comment (shubham-bansal96) [reply pkg/fission-cli/cmd/package/create.go]: yes, that part is already handled in code. If a package is created without spec and have name more than 63 characters then it get validate by k8s, but when package is created using spec then we need to validate it manually. And i have made my changes accordingly.

**Pull request merged**